### PR TITLE
Hubert / branding removal toggle

### PR DESCRIPTION
**Pull request by hubert-deriv**

### CODE REVIEW

File: src/components/homepage/Homepage/HomepageFooter/HomepageFooter.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { useSelector } from "@xstate/react";
+import { stateService } from "../../../../state/stateSignal";
+import { isThirdPartyHostSelector } from '../../../../state/selectors';
+
+export default function HomepageFooter() {
+    const is_third_party_host = useSelector(stateService, isThirdPartyHostSelector);
+    if (is_third_party_host) {
+        return null;
```
Comment (prince-deriv) [inline src/components/homepage/Homepage/HomepageFooter/HomepageFooter.tsx:8]:
```suggestion
        return <></>;
```

File: src/components/global/DerivLogo/DerivLogo.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { Link } from "react-router-dom"
+import styles from './DerivLogo.module.scss';
+import { useSelector } from "@xstate/react";
+import { stateService } from "../../../state/stateSignal";
+import { isThirdPartyHostSelector } from '../../../state/selectors';
+
+export default function DerivLogo() {
+    const is_third_party_host = useSelector(stateService, isThirdPartyHostSelector);
+    if (is_third_party_host) {
+        return null;
```
Comment (prince-deriv) [inline src/components/global/DerivLogo/DerivLogo.tsx:10]:
```suggestion
        return <></>;
```

File: src/components/global/Header/TopNav.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import styles from "./Header.module.scss";
+import { useSelector } from "@xstate/react";
+import { stateService } from "../../../state/stateSignal";
+import { isThirdPartyHostSelector } from '../../../state/selectors';
+export default function TopNav() {
+    const is_third_party_host = useSelector(stateService, isThirdPartyHostSelector);
+    if (is_third_party_host) {
+        return null;
```
Comment (prince-deriv) [inline src/components/global/Header/TopNav.tsx:8]:
```suggestion
        return <></>;
```

**Review by prince-deriv — COMMENTED** [review]
You can also create a helper function like  `renderOfficialDomainContents` 

```
const renderOfficialDomainContents = (component) => {
   const is_third_party_host = useSelector(stateService, isThirdPartyHostSelector);
   
   return  is_third_party_host ? <></> : component
 }

File: src/components/global/Header/TopNav.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import styles from "./Header.module.scss";
+import { useSelector } from "@xstate/react";
+import { stateService } from "../../../state/stateSignal";
+import { isThirdPartyHostSelector } from '../../../state/selectors';
+export default function TopNav() {
```
Comment (matin-binary) [inline src/components/global/Header/TopNav.tsx:5]: let's use ES6 functions

File: src/components/homepage/Homepage/HomepageFooter/HomepageFooter.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+import { useSelector } from "@xstate/react";
+import { stateService } from "../../../../state/stateSignal";
+import { isThirdPartyHostSelector } from '../../../../state/selectors';
+
+export default function HomepageFooter() {
```
Comment (matin-binary) [inline src/components/homepage/Homepage/HomepageFooter/HomepageFooter.tsx:5]: ES6 functions

File: src/components/homepage/Homepage/HomepageFooter/HomepageFooter.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    const is_third_party_host = useSelector(stateService, isThirdPartyHostSelector);
+    if (is_third_party_host) {
+        return null;
+    }
+    
+    return (
+    <div className="container dark" id="derivFooter">
```
Comment (matin-binary) [inline src/components/homepage/Homepage/HomepageFooter/HomepageFooter.tsx:12]: seems like the indentation is off here.

**Comment by hubert-deriv** [comment]
> You can also create a helper function like `renderOfficialDomainContents`
> 
> ```
> const renderOfficialDomainContents = (component) => {
>    const is_third_party_host = useSelector(stateService, isThirdPartyHostSelector);
>    
>    return  is_third_party_host ? <></> : component
>  }
> ```

Great idea, thanks!

**Review by prince-deriv — APPROVED** [review]
LGTM

**Review by matin-binary — APPROVED** [review]
LGTM

**Pull request merged**